internal/handler/wire.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
```

Comment:
About https://github.com/FerretDB/FerretDB/pull/4501#issuecomment-2261816468, I think it's possible to only validate it in handler, upon calling `OpMsgDocument()` and `NewOpMsg()`. Other places should not need to validate document. I think using `types` in `clientconn` can be avoided and should be avoided. Because that conversion is not necessary in my opinion.

The intended usage in handler is like this.
```
func (h *Handler) MsgAggregate(connCtx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
	document, err := OpMsgDocument(msg)
	...
```

```
	return NewOpMsg(
		must.NotFail(types.NewDocument(
			"ok", float64(1),
			...
		)
	)
```

Reply:
Doing this way means it doesn't stop a new handler to use wire message directly like `wire.NewOpMsg(must.NotFail(bson.ConvertBSONDocument(doc)))` instead of calling `newOpMsg(msg)`. If that happens validation can be skipped, we may add linter for that to ensure handler package always calls validation.

Reply:
Not relevant any more, as validation is handled in `wire`